### ISSUE TITLE
(🐞) bad `unnecessary-dunder-call` report for `object.__getattribute__(self` within a `__getattribute__` implementation

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#9486]
```py
class A:
    a = 1

    def __getattribute__(self, item):
        return object.__getattribute__(self, item)  # PLC2801 Unnecessary dunder call to `__getattribute__`. Access attribute directly or use getattr built-in function..

A().a
```
Ah yes, better use `getattr` built-in function..

```py
class A:
    a = 1

    def __getattribute__(self, item):
        return getattr(self, item)

A().a
#   File "C:\Users\charlie_marsh(parody)\rust_based_python_typechecker\__init__.py", line 5, in __getattribute__
#    return getattr(self, item)
#           ^^^^^^^^^^^^^^^^^^^
#  [Previous line repeated 745 more times]
#RecursionError: maximum recursion depth exceeded
```

**Comment by charliermarsh** [comment]
What is going on with that filename lol

**Comment by randolf-scholz** [comment]
Other example of a false-positive:

```python
class A:
    def __repr__(self):
        return "Prints this text"

a = A()
object.__repr__(a)  # PLC2801
```

Generally, for `T.__dunder__(obj, *args, **kwargs)` to be redundant, I think we necessarily need `type(obj).__dunder__ is T.__dunder__` to be true?

**Comment by charliermarsh** [comment]
Need to understand how pylint handles this.

**Comment by charliermarsh** [comment]
So for one thing, they omit dunder calls _within_ dunder methods.

**Comment by charliermarsh** [comment]
It looks like they also skip dunder calls on non-instantiated classes? Which includes `object`?

**Comment by bastimeyer** [comment]
> So for one thing, they omit dunder calls _within_ dunder methods.

This is the main cause of `PLC2801` being raised in our project after updating ruff to `0.1.13`.

Another false positive report (IMO) after upgrading to the latest version is the explicit call of dunder methods in test cases. Imagine a `__getitem__` method which gets tested for a specific exception in a `with pytest.raises(...)` block. Ruff now wants me to use the subscript notation (`foo[bar]` instead of `foo.__getitem__(bar)`), which is then reported as a "statement without an effect" by other linters (like the one built into pycharm for example). Calling the `__getitem__` method explicitly should be ok in test cases and shouldn't require an annotation for suppressing `PLC2801`.

**Comment by charliermarsh** [comment]
Yeah we can easily fix that part. The second part is harder and will require us to greatly limit the scope of the rule.

**Comment by charliermarsh** [comment]
(I'll fix the first part now.)
